embedded/document/engine.go

```diff
@@ -444,55 +444,27 @@ func (e *Engine) upsertDocument(ctx context.Context, collectionName string, doc
 	return docID, txID, rev, nil
 }
 
-func (e *Engine) GetDocuments(ctx context.Context, collectionName string, queries []*Query, pageNum int, itemsPerPage int) ([]*structpb.Struct, error) {
+func (e *Engine) GetDocuments(ctx context.Context, collectionName string, queries []*Query) (sql.RowReader, error) {
```

Comment:
sql row reader should not be exposed but a documentReader can wrap it

Reply:
I'd think limit and offset still applies ...

Reply:
I have added offset, but limit can' think of the use case for it